lib/view_component/base.rb

```diff
@@ -179,6 +179,10 @@ def short_identifier
       end
 
       def inherited(child)
+        # Compile so child will inherit compiled `call_*` template methods that
+        # `compile` defines
+        compile
```

Comment:
Shouldn't compilation be triggered at compile time? That is, when a child compononent is compiled, it should first ensure its parent is compiled.

During development, this would cause more compilations than necessary.

Reply:
@fsateler That's a really great question! If I had to guess, the performance impact would be relatively small. I plan on trying to dig into some numbers to see what that impact may be, then make an informed decision from there.

Reply:
Well, I figure there's a reason why we delay compilation for components in general :wink:

Reply:
I checked the numbers and compiles look like they're super fast. I was seeing the slowest compile take around 1ms on my machine with the majority of compiles being much faster than that. I think that seems like a fine trade-off (at least for now) instead of adding more complexity here.